src/Conversation/conversationManager.ts

```diff
@@ -0,0 +1,183 @@
+import { Conversation } from "../ChatModel/ChatModel";
+import { ExtensionContext } from "vscode";
+import { ChatMessage } from "../ChatModel/ChatModel";
+import { CONTEXT_INSTRUCTION } from "../commands/constants";
+
+export class ConversationManager {
+  // TODO: Use constants where needed
+  private readonly context: ExtensionContext;
+
+  constructor(context: ExtensionContext) {
+    this.context = context;
+  }
+
+  get conversations(): Conversation[] {
+    return (
+      this.context.workspaceState.get<Conversation[]>("conversations") || []
+    );
+  }
+
+  set conversations(value: Conversation[]) {
+    const ids = value.map((conversation) => conversation.id);
+    if (new Set(ids).size !== ids.length) {
+      throw new Error("Duplicate ids");
+    }
+    this.context.workspaceState.update("conversations", value);
+  }
+
+  get conversationIds(): number[] {
+    return this.context.workspaceState.get<number[]>("conversationIds") || [];
+  }
+
+  set conversationIds(value: number[]) {
+    if (new Set(value).size !== value.length) {
+      throw new Error("Duplicate ids");
+    }
+    this.context.workspaceState.update("conversationIds", value);
+  }
+
+  get activeConversationId(): number | null {
+    return (
+      this.context.workspaceState.get<number>("activeConversationId") || null
+    );
+  }
+
+  set activeConversationId(value: number | null) {
+    if (value && !this.conversationIds.includes(value)) {
+      throw new Error("Conversation ID does not exist");
+    }
+    this.context.workspaceState.update("activeConversationId", value);
+  }
+
+  getConversation(id: number): Conversation | undefined {
+    return this.conversations.find((conversation) => conversation.id === id);
```

Comment:
Would it be better to use a map to hold conversations than an array? That way we can just do `this.conversations.get(id)`.

Reply:
From what I understand, the workspace state used here requires that the objects saved be JSON serializable, which `Map`s are for some reason not. There are some different ways around this that I haven't looked all that much into, but this array approach was working fine and the data we are storing isn't all that complex so I didn't spend time figuring it out. Might not be hard and might technically be the better choice, so I'm happy to look into using a `Map` instead if we want to move that over.

Reply:
That sounds like a good reason to me. If it's working, then don't worry about it.